src/pages/patientView/sampleManager.tsx

```diff
@@ -122,21 +123,33 @@ class SampleManager {
         // remove common CANCER_TYPE/CANCER_TYPE_DETAILED in top bar (display on
         // patient)
         ['CANCER_TYPE', 'CANCER_TYPE_DETAILED'].forEach((attr) => {
-            if (SampleManager.isSameClinicalAttributeInAllSamples(samples, attr)) {
-                this.commonClinicalDataLegacyCleanAndDerived[attr] = this.clinicalDataLegacyCleanAndDerived[samples[0].id][attr];
-                samples.forEach((sample) => {
+            if (SampleManager.isSameClinicalAttributeInAllSamples(_samples, attr)) {
+                this.commonClinicalDataLegacyCleanAndDerived[attr] = this.clinicalDataLegacyCleanAndDerived[_samples[0].id][attr];
+                _samples.forEach((sample) => {
                     delete this.clinicalDataLegacyCleanAndDerived[sample.id][attr];
                 });
             }
         })
 
-        this.samples = sortSamples(samples, this.clinicalDataLegacyCleanAndDerived, events);
-        this.samples.forEach((sample, i) => {
-            this.sampleIndex[sample.id] = i;
-            this.sampleLabels[sample.id] = String(i+1);
-        });
-        // order as array of sample ids (used further downstream)
-        this.sampleOrder = _.sortBy(Object.keys(this.sampleIndex), (k) => this.sampleIndex[k]);
+        this._samples = sortSamples(_samples, this.clinicalDataLegacyCleanAndDerived, events);
+    }
+
+    @computed public get samples() {
+        const urlSampleOrder = JSON.parse(this.urlWrapper.query.sampleIdOrder || "[]");
```

Comment:
i have that we do the parsing on as needed basis.  i think it should take place in one place.  i don't think SampleManager should be aware of url